app/packages/registration-schema/page-manager.ts

```diff
@@ -0,0 +1,40 @@
+import Changeset from 'ember-changeset';
+import lookupValidator from 'ember-changeset-validations';
+import { ChangesetDef } from 'ember-changeset/types';
+import {
+    buildValidation,
+    getSchemaBlockGroups,
+    ResponseValue,
+    SchemaBlock,
+    SchemaBlockGroup,
+} from 'ember-osf-web/packages/registration-schema';
+import { PageResponse } from 'ember-osf-web/packages/registration-schema/page-response';
+
+export class PageManager {
+    changeset?: ChangesetDef;
+    schemaBlockGroups?: SchemaBlockGroup[];
+    pageResponses?: Record<string, ResponseValue>;
+    pageHeadingText?: string;
+
+    constructor(pageSchemaBlocks: SchemaBlock[], registrationResponses: PageResponse) {
+        this.schemaBlockGroups = getSchemaBlockGroups(pageSchemaBlocks);
+        this.pageHeadingText = this.schemaBlockGroups[0].labelBlock!.displayText;
+        this.pageResponses = {};
+        for (const group of this.schemaBlockGroups) {
+            if (group.registrationResponseKey) {
+                this.pageResponses[group.registrationResponseKey]
+                    = registrationResponses[group.registrationResponseKey];
+            }
+        }
+        const validations = buildValidation(this.schemaBlockGroups);
+        this.changeset = new Changeset(this.pageResponses, lookupValidator(validations), validations) as ChangesetDef;
+    }
+
+    pageIsValid() {
+        if (this.changeset) {
+            this.changeset.validate();
```

Comment:
Hmm, so running validations is a side effect of checking validation state? That seems wrong --  checking state should never have side effects. We should run validations explicitly, in response to user action.

As i recall, we want two properties (either implemented as getters or computed props): `isValid` and `isVisited`. `isValid` just reflects `changeset.isValid`. I'm not sure we nailed down the details of `isVisited`, but we talked about using the presence of keys in `registrationResponses` -- if any of the page's response keys are present (even if the value is `undefined`), we can consider the page visited.

How about let's compute `isVisited` in the constructor, and let `DraftRegistrationManager` update it when visiting a new page?